{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "from matplotlib import pyplot as plt\n",
    "from mpl_toolkits.mplot3d import Axes3D\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import re\n",
    "import seaborn as sns\n",
    "\n",
    "#from google.colab import files\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE\n",
    "from sklearn.preprocessing import LabelEncoder, OneHotEncoder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"1-ngs_raw_dataset.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>coding_change</th>\n",
       "      <th>protein_change</th>\n",
       "      <th>allele_frequency</th>\n",
       "      <th>read_depth</th>\n",
       "      <th>gene_refgene</th>\n",
       "      <th>exonic_func_refgene</th>\n",
       "      <th>func_refgene</th>\n",
       "      <th>1000genomes</th>\n",
       "      <th>vest3_rankscore</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>database_classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>7902</td>\n",
       "      <td>chr13</td>\n",
       "      <td>32911756</td>\n",
       "      <td>32911756</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>c.3264T&gt;C</td>\n",
       "      <td>p.(Pro1088Pro)</td>\n",
       "      <td>0.428000</td>\n",
       "      <td>1189</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>synonymous_SNV</td>\n",
       "      <td>exonic</td>\n",
       "      <td>0.006789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>7902</td>\n",
       "      <td>chr13</td>\n",
       "      <td>32913910</td>\n",
       "      <td>32913910</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>c.5418A&gt;G</td>\n",
       "      <td>p.(Glu1806Glu)</td>\n",
       "      <td>0.475531</td>\n",
       "      <td>2023</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>synonymous_SNV</td>\n",
       "      <td>exonic</td>\n",
       "      <td>0.006789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>7902</td>\n",
       "      <td>chr13</td>\n",
       "      <td>32929007</td>\n",
       "      <td>32929007</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>c.7017G&gt;C</td>\n",
       "      <td>p.(Lys2339Asn)</td>\n",
       "      <td>0.554974</td>\n",
       "      <td>573</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>nonsynonymous_SNV</td>\n",
       "      <td>exonic</td>\n",
       "      <td>0.006789</td>\n",
       "      <td>0.380</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>7902</td>\n",
       "      <td>chr13</td>\n",
       "      <td>32944627</td>\n",
       "      <td>32944627</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>c.8420C&gt;T</td>\n",
       "      <td>p.(Ser2807Leu)</td>\n",
       "      <td>0.020000</td>\n",
       "      <td>1786</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>nonsynonymous_SNV</td>\n",
       "      <td>exonic</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.717</td>\n",
       "      <td>ID=COSM6612534,COSM6612533;OCCURENCE=2(large_i...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>7902</td>\n",
       "      <td>chr13</td>\n",
       "      <td>32972380</td>\n",
       "      <td>32972380</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>c.9730G&gt;A</td>\n",
       "      <td>p.(Val3244Ile)</td>\n",
       "      <td>0.658798</td>\n",
       "      <td>932</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>nonsynonymous_SNV</td>\n",
       "      <td>exonic</td>\n",
       "      <td>0.006789</td>\n",
       "      <td>0.366</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample chromosome     start       end ref alt coding_change  \\\n",
       "0    7902      chr13  32911756  32911756   T   C     c.3264T>C   \n",
       "1    7902      chr13  32913910  32913910   A   G     c.5418A>G   \n",
       "2    7902      chr13  32929007  32929007   G   C     c.7017G>C   \n",
       "3    7902      chr13  32944627  32944627   C   T     c.8420C>T   \n",
       "4    7902      chr13  32972380  32972380   G   A     c.9730G>A   \n",
       "\n",
       "   protein_change  allele_frequency  read_depth gene_refgene  \\\n",
       "0  p.(Pro1088Pro)          0.428000        1189        BRCA2   \n",
       "1  p.(Glu1806Glu)          0.475531        2023        BRCA2   \n",
       "2  p.(Lys2339Asn)          0.554974         573        BRCA2   \n",
       "3  p.(Ser2807Leu)          0.020000        1786        BRCA2   \n",
       "4  p.(Val3244Ile)          0.658798         932        BRCA2   \n",
       "\n",
       "  exonic_func_refgene func_refgene  1000genomes  vest3_rankscore  \\\n",
       "0      synonymous_SNV       exonic     0.006789              NaN   \n",
       "1      synonymous_SNV       exonic     0.006789              NaN   \n",
       "2   nonsynonymous_SNV       exonic     0.006789            0.380   \n",
       "3   nonsynonymous_SNV       exonic     0.000000            0.717   \n",
       "4   nonsynonymous_SNV       exonic     0.006789            0.366   \n",
       "\n",
       "                                            cosmic70                  CLNSIG  \\\n",
       "0                                                NaN                  Benign   \n",
       "1                                                NaN                  Benign   \n",
       "2                                                NaN                  Benign   \n",
       "3  ID=COSM6612534,COSM6612533;OCCURENCE=2(large_i...  Uncertain_significance   \n",
       "4                                                NaN                  Benign   \n",
       "\n",
       "   database_classification  \n",
       "0                        5  \n",
       "1                        5  \n",
       "2                        5  \n",
       "3                        0  \n",
       "4                        5  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
